internal.nix

```diff
@@ -33,6 +33,26 @@ rec {
   makeValidDrvName = str:
     lib.stringAsChars (c: if isValidDrvNameChar c then c else "?") str;
 
+  # Description: Checks if a string looks like a valid git revision
+  # Type: String -> Boolean
+  isGitRev = str:
+    (builtins.match "[0-9a-f]{40}" str) != null;
+
+  # Description: Takes a string of the format "git+http(s)://domain.tld/repo#commitish" and returns
+  # an attribute set { url, commitish }
+  # Type: String -> Set
+  parseGitRef = str:
+    let
+      parts = builtins.split "[#]" str;
```

Comment:
Why do we need the `[` & `]` in the pattern? Wouldn't splitting at just `#` be enough?